arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2893,6 +2920,319 @@ void RestReplicationHandler::handleCommandLoggerTickRanges() {
   }
 }
 
+bool RestReplicationHandler::prepareRevisionOperation(RevisionOperationContext& ctx) {
+  auto& selector = server().getFeature<EngineSelectorFeature>();
+  if (!selector.isRocksDB()) {
+    generateError(
+        rest::ResponseCode::NOT_IMPLEMENTED, TRI_ERROR_NOT_IMPLEMENTED,
+        "this storage engine does not support revision-based replication");
+    return false;
+  }
+
+  LOG_TOPIC("253e2", TRACE, arangodb::Logger::REPLICATION)
+      << "enter prepareRevisionOperation";
+
+  bool found = false;
+
+  // get collection Name
+  ctx.cname = _request->value("collection");
+  if (ctx.cname.empty()) {
+    generateError(rest::ResponseCode::BAD, TRI_ERROR_HTTP_BAD_PARAMETER,
+                  "invalid collection parameter");
+    return false;
+  }
+
+  // get batchId
+  std::string const& batchIdString = _request->value("batchId", found);
+  if (found) {
+    ctx.batchId = StringUtils::uint64(batchIdString);
+  } else {
+    generateError(rest::ResponseCode::NOT_FOUND, TRI_ERROR_HTTP_BAD_PARAMETER,
+                  "replication revision tree - request misses batchId");
+    return false;
+  }
+
+  // get resume
+  std::string const& resumeString = _request->value("resume", found);
+  if (found) {
+    ctx.resume = StringUtils::uint64(resumeString);
```

Comment:
does this throw if it fails?

Reply:
So we have an implementation which utilizes `std::from_chars`, which is non-throwing. Currently the implementation seems to rely on a potentially fragile detection method, but since we're on C++17 officially now, we can just make it default. Will adjust.